python_modules/libraries/dagster-aws/dagster_aws/ecs/client.py

```diff
@@ -1,259 +1,138 @@
-        basedict = {
-            "executionRoleArn": "arn:aws:iam::{}:role/ecsTaskExecutionRole".format(self.account_id),
-            "containerDefinitions": [
-                {
-                    "logConfiguration": {
-                        "logDriver": "awslogs",
-                        "options": {
-                            "awslogs-group": "/ecs/{}".format(fmtstr),
-                            "awslogs-region": "us-east-2",
-                            "awslogs-stream-prefix": "ecs",
-                        },
-                    },
-                    "entryPoint": entrypoint,
-                    "portMappings": [{"hostPort": 8080, "protocol": "tcp", "containerPort": 8080}],
-                    "command": command,
-                    "cpu": 0,
-                    "environment": [{"name": "TEST", "value": "end"}],
-                    "image": imagename,
-                    "name": containername,
-                }
-            ],
-            "placementConstraints": [],
-            "memory": memory,
-            "family": family,
-            "requiresCompatibilities": ["FARGATE"],
-            "networkMode": "awsvpc",
-            "cpu": cpu,
-        }
-        response = self.ecs_client.register_task_definition(**basedict)
-        arn = response["taskDefinition"]["taskDefinitionArn"]
-        self.task_definition_arn = arn
-        self.container_name = containername
-
-    def run_task(self, **kwargs):
-        """
-        sets this client's task running.
+            task_definition (str): The family and revision (family:revision) or full ARN
+                of the task definition to run. If a revision is not specified, the
+                latest ACTIVE revision is used.
 
-        TODO: explicitly add overrides as params here for cpu, command, and environment vars
-
-        Args:
-            **kwargs: Any add'l params you want the task to have
+        Returns:
+            None
         """
-        taskdict = self.ecs_client.run_task(
-            taskDefinition=self.task_definition_arn,
-            launchType=self.launch_type,
+        response = self.client.run_task(
+            count=1,
+            launchType="FARGATE",
+            taskDefinition=task_definition,
             cluster=self.cluster,
-            **kwargs,
+            networkConfiguration={
+                "awsvpcConfiguration": {"subnets": self.subnets, "assignPublicIp": "ENABLED"}
+            },
         )
-        task = taskdict["tasks"][0]
-        container = task["containers"][0]
-        running_task_arn = container["taskArn"]
-        self.tasks.append(running_task_arn)
-        self.tasks_done.append(False)
-        self.offset += 1
+        task_arns = [task["taskArn"] for task in response["tasks"]]
 
-    def check_if_done(self, offset=0):
-        """
-        polls ECS to see if the task is complete, pulling down logs if so
+        retries = self.max_polls
+        while retries > 0:
```

Comment:
`ECSClient.run_task` implements its own polling instead of using the TaskStopped Waiter:

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Waiter.TasksStopped

This is because I couldn't get the Waiter to play nicely with botocore's `Stubber`; it passes a `waiterConfig` param to the DescribeTasks endpoint but the `Stubber` doesn't accept that as a valid param.